### ISSUE TITLE
Incorrect warning on ~ in PATH

### ISSUE THREAD

**Issue by fclaerho** [ansible/ansible-lint#3182]
##### Summary

On running ansible-lint, with PATH containing a name with ~ in it  (such as `/Users/jdoe/Library/Mobile Documents/com~apple~CloudDocs` on MacOS), we get the incorrect warning that:

`WARNING: PATH altered to expand ~ in it. Read https://stackoverflow.com/a/44704799/99834 and correct your system configuration.`

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
WARNING: PATH altered to expand ~ in it. Read https://stackoverflow.com/a/44704799/99834 and correct your system configuration.
Unable to parse ansible cli version: ansible 2.9.27
  config file = None
  configured module search path = ['/Users/jdoe/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/jdoe/bin/.ansible/ansible
  executable location = /Users/jdoe/bin/.ansible/bin/ansible
  python version = 3.8.2 (default, Apr  8 2021, 23:19:18) [Clang 12.0.5 (clang-1205.0.22.9)]

Keep in mind that only 2.12 or newer are supported.
```

- Platform: MacOS 13.x

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

Something like:
```
PATH=$PATH:my~path ansible-lint --version
```

##### Desired Behavior

No warning should be reported.

Given the entry in PATH does not start with ~, but contains ~, it should not be interpolated by ansible-lint as it is done in the function `path_inject()` which does not check the ~ is at the start:
```
    for idx, path in enumerate(paths):
        if "~" in path:  # pragma: no cover
            paths[idx] = os.path.expanduser(path)
            expanded = True
```

##### Actual Behavior

An incorrect warning is reported.

**Comment by ssbarnea** [comment]
@fclaerho I think that is your opportunity to make your first contribution, this can be fixed with one line, `path.startswith("~")` instead of "in" test....
